src/faq/plugin.ts

```diff
@@ -31,301 +29,330 @@ const faqExtension: JupyterLabPlugin<void> = {
   autoStart: true
 };
 
+/**
+ * The class name added to the FAQ plugin.
+ */
+const FAQ_CLASS = 'jp-FAQ';
 
 /**
- * A widget that displays FAQ information.
+ * The id name added to the header section element.
  */
-class FAQWidget extends Widget {
-  /**
-   * A handler for `after-attach` messages.
-   */
-  protected onAfterAttach(msg: Message): void {
-    let answerNodeList = this.node.querySelectorAll('li.jp-FAQ-answer');
-    let answerArray = Array.prototype.slice.call(answerNodeList);
-    for (let answerElement of answerArray) {
-      answerElement.className += ' jp-FAQ-hide';
-    }
-    $('li.jp-FAQ-question').click(function(){
-      $(this).data('clicked', false).next().slideToggle(200)
-      .siblings('li.jp-FAQ-answer').slideUp(200);
-    });
+const HEADER_ID = 'faq-header';
+
+/**
+ * The class name added to the title.
+ */
+const TITLE_CLASS = 'jp-FAQ-title';
+
+/**
+ * The class name added to h1 elements.
+ */
+const HEADER_CLASS = 'jp-FAQ-h1';
+
+/**
+ * The class name added to h2 elements.
+ */
+const SUBHEADER_CLASS = 'jp-FAQ-h2';
+
+/**
+ * The class name added for the question mark icon from default-theme.
+ */
+const QUESTIONMARK_ICON_CLASS = 'jp-QuestionMark';
+
+/**
+ * The class named added the question mark icon.
+ */
+const QUESTIONMARK_CLASS = 'jp-FAQ-QuestionMark';
+
+/**
+ * The class name added to faq content.
+ */
+const CONTENT_CLASS = 'jp-FAQ-content';
+
+/**
+ * The class name added to unordered list elements.
+ */
+const FAQ_LIST_CLASS = 'jp-FAQ-ul';
+
+/**
+ * The class name added to table of contents elements.
+ */
+const TOC_CLASS = 'jp-FAQ-toc';
+
+/**
+ * The class name added to questions.
+ */
+const QUESTION_CLASS = 'jp-FAQ-question';
+
+/**
+ * The class name added to answers.
+ */
+const ANSWER_CLASS = 'jp-FAQ-answer';
+
+/**
+ * The class name added to anchor elements.
+ */
+const ANCHOR_CLASS = 'jp-FAQ-a';
+
+
+class FaqModel extends VDomModel {
```

Comment:
@charnpreetsingh, I think with some added docstrings and making the attributes public like the previous PR this should be ready.